From 8b9c0d1e2f3a4b5c6d7e8f9a0b1c2d3e4f5a6b7 Mon Sep 17 00:00:00 2001
From: Anaconda Recipes <build@anaconda.com>
Date: Wed, 10 Sep 2026 17:06:00 -0600
Subject: [PATCH 0046] Register pybind11_protobuf protobuf7 patch (osx)

Wire 0002 into the pybind11_protobuf tf_http_archive on osx (linux uses 0024).
---
 tensorflow/workspace2.bzl      | 1 +
 third_party/xla/workspace2.bzl | 1 +
 2 files changed, 2 insertions(+)

diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
--- a/tensorflow/workspace2.bzl
+++ b/tensorflow/workspace2.bzl
@@ -810,6 +810,7 @@ def _tf_repositories():
         urls = tf_mirror_urls("https://github.com/pybind/pybind11_protobuf/archive/f02a2b7653bc50eb5119d125842a3870db95d251.zip"),
         sha256 = "3cf7bf0f23954c5ce6c37f0a215f506efa3035ca06e3b390d67f4cbe684dce23",
         strip_prefix = "pybind11_protobuf-f02a2b7653bc50eb5119d125842a3870db95d251",
+        patch_file = ["//third_party/pybind11_protobuf:0002-protobuf7-descriptor-database-string-view.patch"],
     )
 
     tf_http_archive(
diff --git a/third_party/xla/workspace2.bzl b/third_party/xla/workspace2.bzl
--- a/third_party/xla/workspace2.bzl
+++ b/third_party/xla/workspace2.bzl
@@ -503,6 +503,7 @@ def _tf_repositories():
         urls = tf_mirror_urls("https://github.com/pybind/pybind11_protobuf/archive/f02a2b7653bc50eb5119d125842a3870db95d251.zip"),
         sha256 = "3cf7bf0f23954c5ce6c37f0a215f506efa3035ca06e3b390d67f4cbe684dce23",
         strip_prefix = "pybind11_protobuf-f02a2b7653bc50eb5119d125842a3870db95d251",
+        patch_file = ["//third_party/pybind11_protobuf:0002-protobuf7-descriptor-database-string-view.patch"],
     )
 
     java_import_external(
